### PR TITLE
feat(LOC-2289-2463): Revert images from backups

### PR DESCRIPTION
**Pull request by veryspry**
### Summary

We want to give users a way to restore images from backups, so here we go. Some notable items:

- Adds new `main` thread function that restores an image from backup (plus tests for this)
- Adds new table under the "overview" section that shows all successfully compressed and errored images
- Adds a new context menu item to that table that adds the option to restore the image from backup (this is only added if `compressedImageHash` is present on the associated image object
- Refactored some of the table related React components to utilize them for this plus their original intended use
- Adds a confirm restore from backup modal

### Screenshots

Screen Shot 2020-12-09 at 11 11 57 AM

Screen Shot 2020-12-09 at 11 12 14 AM

Screen Shot 2020-12-09 at 11 12 19 AM

Screen Shot 2020-12-09 at 11 12 03 AM

### Reference
- [LOC-2289](https://getflywheel.atlassian.net/browse/LOC-2289)
- [LOC-2463](https://getflywheel.atlassian.net/browse/LOC-2463)

### CODE REVIEW

**Comment by veryspry** [comment]
Looks like we have a request to add analytics tracking to this. I'll have a PR incoming to address that, but that shouldn't change the existing code much, if at all.

**Comment by veryspry** [comment]
> Looks like we have a request to add analytics tracking to this. I'll have a PR incoming to address that, but that shouldn't change the existing code much, if at all.

Analytics tracking items added!

File: src/main/createStore.ts [hunk 1]
```diff
@@ -63,6 +67,29 @@ export function createStore(initialState: SiteDataBySiteID = {}): Store {
...
 		},
+
+		/**
+		 * Merges in new image data for a single image associated with a site
+		 *
+		 * @param siteID
+		 * @param imageID
+		 * @param newState
+		 */
+		setStateByImageID(siteID, imageID, newState) {
```
Comment (andrewbotz) [inline src/main/createStore.ts:78]: Minor but I wonder about the consistency of our naming conventions -- `siteId` Vs `siteID` and `imageID` vs `imageId`

Comment (veryspry) [reply src/main/createStore.ts:78]: Same over here actually. I think I prefer `siteId`, but for some reason, went with `siteID` in the early days of the repo and never changed it 😬

**Review by andrewbotz — APPROVED** [review]
Tested and everything seemed to work well.

**Pull request merged**